client/src/utils/calendarUtils.js

```diff
@@ -0,0 +1,69 @@
+import moment from "moment";
+
+export const getTimeSlots = (
+  freeTimeInterval,
+  meetingLength,
+  availabilityTimes
+) => {
+  const timeSlots = freeTimeInterval;
+  const result = timeSlots.map(item => breakUpTimes(item, meetingLength));
+  
+  //filter out unavailable times
+  const filteredTimes = removeUnavailableTimes(
+    result.flat(),
+    availabilityTimes
+  );
+  return filteredTimes;
+};
+
+const milliscondsToMinutes = milliseconds => {
+  return Math.floor(milliseconds / 60000);
+};
+
+const breakUpTimes = (timeslot, meetingLength) => {
+  const intervals = [];
+  const startTime = timeslot.start;
+  const endTime = moment(timeslot.end);
+  const meetingEndTime = moment(startTime);
+
+  const intervalLength = meetingLength * -1;
+
+  while (milliscondsToMinutes(meetingEndTime.diff(endTime)) <= intervalLength) {
+    const meetingStartHour = meetingEndTime.get("hours");
+    const meetingStartMinute = meetingEndTime.get("minutes");
+    const minutesFormated =
+      meetingStartMinute < 10
+        ? "0" + meetingStartMinute
+        : meetingStartMinute + "";
+    intervals.push(`${meetingStartHour}:${minutesFormated}`);
```

Comment:
You might be able to use some date-fns methods to format the date easier since we're already importing it for other parts of the app
https://date-fns.org/v2.16.1/docs/formatDuration